packaging/tags.py

```diff
@@ -26,14 +24,39 @@
 _32_BIT_INTERPRETER = sys.maxsize <= 2 ** 32
 
 
-@attr.s(frozen=True, repr=False)
 class Tag(object):
-    interpreter = attr.ib(converter=str.lower)
-    abi = attr.ib(converter=str.lower)
-    platform = attr.ib(converter=str.lower)
+
+    __slots__ = ["_interpreter", "_abi", "_platform"]
+
+    def __init__(self, interpreter, abi, platform):
+        self._interpreter = str.lower(interpreter)
+        self._abi = str.lower(abi)
+        self._platform = str.lower(platform)
```

Comment:
should these be `interpreter.lower()` / `abi.lower()` / etc.

Reply:
@asottile it wouldn't hurt, but it technically isn't wrong. :) I'll quickly fix it separately.